CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+# Contributing to rust_codegen_gcc
+
+Welcome to the rust_codegen_gcc project! This guide will help you get started as a contributor. The project aims to provide a GCC codegen backend for rustc, allowing Rust compilation on platforms unsupported by LLVM and potentially improving runtime performance through GCC's optimizations.
+
+## Getting Started
+
+### Setting Up Your Development Environment
+
+1. Install the required dependencies:
+   - rustup (follow instructions on the [official website](https://rustup.rs))
+   - DejaGnu (for running libgccjit test suite)
+   - Additional packages: `flex`, `libmpfr-dev`, `libgmp-dev`, `libmpc3`, `libmpc-dev`
+
+2. Clone and configure the repository:
+   ```bash
+   git clone https://github.com/rust-lang/rust_codegen_gcc
+   cd rust_codegen_gcc
+   cp config.example.toml config.toml
+   ```
+
+3. Build the project:
+   ```bash
+   ./y.sh prepare  # downloads and patches sysroot
+   ./y.sh build --sysroot --release
+   ```
+
+### Running Tests
+
+To verify your setup:
+```bash
+# Run the full test suite
+./y.sh test --release
+
+# Test with a simple program
+./y.sh cargo build --manifest-path tests/hello-world/Cargo.toml
+```
+
+## Communication Channels
+
+- Matrix: Join our [Matrix channel](https://matrix.to/#/#rustc_codegen_gcc:matrix.org)
+- IRC: Join us on [IRC](https://web.libera.chat/#rustc_codegen_gcc)
+- GitHub Issues: For bug reports and feature discussions
+
+We encourage new contributors to join our communication channels and introduce themselves. Feel free to ask questions about where to start or discuss potential contributions.
+
+## Understanding Core Concepts
+
+### Project Structure
+
+The project consists of several key components:
+- The GCC backend integration through libgccjit
+- Rust compiler interface
+- Test infrastructure
+
+### Common Development Tasks
+
+#### Running Specific Tests
+To run a specific test:
+1. Individual test: `./y.sh test --test <test_name>`
```

Comment:
I don't believe this would work that way. The test flag could be `--test-libcore` or `--std-tests` among other things.